action.yml

```diff
@@ -0,0 +1,53 @@
+name: 'Create Pull Request'
+description: 'Create a Pull Request'
+inputs:
+  branch-name:
+    description: 'Branch name'
+    required: true
+  commit-message:
+    description: 'Commit message'
+    required: true
+  title:
+    description: 'PR title'
+    required: false
+    default: ${{ inputs.commit-message }}
+  description:
+    description: 'PR description'
+    required: false
+    default: ''
+  draft:
+    description: 'Create a Draft PR'
+    required: false
+    default: 'false'
+  token:
+    description: 'Personal Access Token'
+    required: false
+    default: ${{ github.token }}
+outputs:
+  pr-url:
+    description: "New PR URL"
+    value: ${{ env.pr-url }}
+runs:
+  using: "composite"
+  steps:
+    - name: Setup
+      run: |  
+        git checkout -b ${{ inputs.branch-name }}
+        git config user.email "${{ github.actor }}@users.noreply.github.com"
+        git config user.name "${{ github.actor }}"
+        git commit -am ${{ inputs.commit-message }}
+        git push -u origin ${{ inputs.branch-name }}
+        echo 'module_pr_url<<EOF' >> $GITHUB_ENV
+      shell: bash
+    - name: Create Draft PR
+      run: gh pr create --title ${{ inputs.title }}  --body ${{ inputs.description }} --draft >> $GITHUB_ENV
```

Comment:
Is there a way to make `--draft` conditional based on `inputs.draft`?

Reply:
Hmm... can you just add `if: inputs.draft == 'true'` and `if inputs.draft != 'true'` to each of these steps?

Reply:
I wonder if breaking the env variable assignment across steps will cause issues? Another option might be having two initial steps: `echo 'draft=' >>$GITHUB_ENV` and `if: inputs.draft == 'true', echo 'draft=--draft' >> $GITHUB_ENV` and then just include $DRAFT in the pr command?

Reply:
I tried that but my IDE was throwing "Property `if` is not allowed".  Maybe a false positive?

Reply:
Ah. Well you can probably just move that into bash then:

```
run: |
        echo 'draft= ' > $GITHUB_ENV
        if [ ${{ inputs.draft }} = 'true' ]; then
          echo 'draft=--draft' > $GITHUB_ENV
        fi
        git checkout -b ${{ inputs.branch-name }}
        git config user.email "${{ github.actor }}@users.noreply.github.com"
        git config user.name "${{ github.actor }}"
        git commit -am ${{ inputs.commit-message }}
        git push -u origin ${{ inputs.branch-name }}
        echo 'module_pr_url<<EOF' >> $GITHUB_ENV
        gh pr create --title ${{ inputs.title }} --body ${{ inputs.description }} $DRAFT >> $GITHUB_ENV
        echo 'EOF' >> $GITHUB_ENV
```